### ISSUE TITLE
Bug: no-lone-blocks emits an error with `using` keyword

### ISSUE THREAD

**Issue by MillerSvt** [eslint/eslint#18204]
### Environment

Node version: 20.11.1
npm version: 10.2.4
Local ESLint version: 8.57.0
Global ESLint version: 8.57.0
Operating System: Mac

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```
{
  rules: {
    'no-lone-blocks': 'error'
  }
}
```
</details>

```js
test('some test', () => {
    // some logic before
    {
        using mock = createMocker('foo');
        fn();
        // some expectations
    }
    // some logic after
});  
```

### What did you expect to happen?

No errors, because using keyword equals const keyword

### What actually happened?

Error: Nested block is redudant

### Link to Minimal Reproducible Example

[Reproduce](https://typescript-eslint.io/play/#ts=5.3.3&fileType=.tsx&code=FAb2AJwVwZwSwHYHNwENwF5wGMBOBTVAF3wFkB7bAa31wAoByAM3PIYEoBuYAXyA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQHYHsBaefXRQgIxIGMBrVDSRaafaScMAXxC6A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by Tanujkanti4441** [comment]
Hi @MillerSvt, thanks for the issue, if i am not wrong `using` keyword is more inclined with typescript right now, so you should open this issue to [typescript-eslint](https://github.com/typescript-eslint/typescript-eslint).

**Comment by Rec0iL99** [comment]
I think so too. Closing.

**Comment by Josh-Cena** [comment]
This should be in ESLint core. `using` is a JS thing, not a TS thing. It's the same as `class`, `async`, or any other syntax—the fact that TS provides transpiling support doesn't mean it's TS-only.

**Comment by mdjermanovic** [comment]
This is a new syntax from the [Explicit Resource Management](https://github.com/tc39/proposal-explicit-resource-management) proposal, which is currently in stage 3 status. Per our [policies about experimental features](https://github.com/eslint/eslint?tab=readme-ov-file#what-about-experimental-features), we'll update core rules to support this syntax when the proposal reaches stage 4. Until then, a solution can be that plugins related to parsers that already support this syntax provide replacement rules.

**Comment by Josh-Cena** [comment]
@mdjermanovic In https://github.com/typescript-eslint/typescript-eslint/issues/8682, we discovered that ESLint can be future proof simply by switching the check to `if (node.kind !== "var")`. Is there desire to make this change?

**Comment by mdjermanovic** [comment]
> switching the check to `if (node.kind !== "var")`

I think that would be fine. @eslint/eslint-team thoughts?

**Comment by nzakas** [comment]
Makes sense to me. :+1:

**Comment by nzakas** [comment]
@MillerSvt @Josh-Cena would either of you like to submit a pull request for this change?

**Comment by Josh-Cena** [comment]
Maybe @kirkwaiblinger too, who first proposed this?

**Comment by nzakas** [comment]
Open to anyone who wants it. :)
